GoogleForCreators/web-stories-wp
# Expanded canvas sizing

## Context

This expands canvas sizing after the page menu has been moved and the carousel has been made collapsible.

**WIP**: This is towards #9724 until that PR is merged. Besides that, this is a complete PR, but shouldn't be merged before #9724 is. Full review is very welcome.

## Relevant Technical Choices
* I tried to organize some of the padding and spacing choices made in the footer a bit better. Centralized the margins and the gaps in particular.

## User-facing changes

| Zoom | Carousel | Before | After |
|-|-|-|-|
| Fit | Expanded | ![before_fit_expanded](https://user-images.githubusercontent.com/637548/142297449-800dc01b-b636-4323-a917-a6dcf1914879.png) | ![after_fit_expanded](https://user-images.githubusercontent.com/637548/142297458-2dab4446-d1fb-4fdb-bf27-d015a7f9aa98.png) |
| Fit | Collapsed | ![before_fit_collapsed](https://user-images.githubusercontent.com/637548/142297518-88ca8da1-795f-4926-a754-69bae853e1f5.png) | ![after_fit_collapsed](https://user-images.githubusercontent.com/637548/142297523-feabedb1-e77f-4cda-b7af-d20a010d6612.png) |
| Zoom | Expanded | ![before_zoom_expanded](https://user-images.githubusercontent.com/637548/142297529-559814c6-00c9-4762-a24c-777dc65f594e.png) | ![after_zoom_expanded](https://user-images.githubusercontent.com/637548/142297536-fc061a7f-9e00-4809-81ea-107359aff351.png) |
| Zoom | Collapsed | ![before_zoom_collapseed](https://user-images.githubusercontent.com/637548/142297540-5d2045b9-e4f6-4ee9-8033-6ff35a166393.png) | ![after_zoom_collapsed](https://user-images.githubusercontent.com/637548/142297549-b1b69a41-0fc8-461b-8461-8a8a7cd222c5.png) |

Note in particular how the bottom margin was basically ignored while zoomed before (the buttons and carousel are flush with the bottom of the window), but that's fixed now.

Also see this other edge-case where the zoom is set to fit and the viewport approximately matches the page dimensions - there was not enough room for the right-side page menu before, but there is now:

| Before | After |
|-|-|
| ![before_fit_narrow](https://user-images.githubusercontent.com/637548/142297312-6357c198-5335-4d19-92d3-b3d22b616b44.png) | ![after_fit_narrow](https://user-images.githubusercontent.com/637548/142297320-73764e30-e908-4153-8f45-b76b0dc98e83.png) | 

## Testing Instructions

This PR can be tested by following these steps:

1. Expand and collapse the carousel, change the zoom setting, and observe the canvas resizing to use as much space as possible.

## Checklist




---

Fixes #9642
